ostd/src/arch/x86/kernel/apic/ioapic.rs

```diff
@@ -190,13 +193,22 @@ pub fn init() {
         return;
     }
     let table = ACPI_TABLES.get().unwrap().lock();
-    let platform_info = PlatformInfo::new(&*table).unwrap();
+    let platform_info = PlatformInfoWrapper::new(&*table).unwrap().0;
     match platform_info.interrupt_model {
         acpi::InterruptModel::Unknown => panic!("not found APIC in ACPI Table"),
         acpi::InterruptModel::Apic(apic) => {
             let mut vec = Vec::new();
             for id in 0..apic.io_apics.len() {
                 let io_apic = apic.io_apics.get(id).unwrap();
+                #[cfg(feature = "cvm_guest")]
+                // SAFETY:
+                // This is safe because we are ensuring that the `io_apic.address` is a valid MMIO address before this operation.
+                // We are also ensuring that we are only unprotecting a single page.
+                if tdx_is_enabled() {
+                    unsafe {
+                        tdx_guest::unprotect_gpa_range(io_apic.address as usize, 1).unwrap();
```

Comment:
So a workaround is needed because the upstream `apic` crate does not expose the `address` field?

Reply:
In fact, we just need a [tiny change](https://github.com/rust-osdev/acpi/commit/26e92e9d65f591b1ab3b8bb358779ba14cf90d8e) and it is merged into `acpi` crate now. But there seem to be new bugs in the latest acpi crate. I am trying to fix them and upgrade the `acpi` crate version.